two_factor/middleware/enforce2fa.py

```diff
@@ -0,0 +1,21 @@
+"""Middleware to check if staff user has 2FA unabled."""
+
+from django.shortcuts import redirect
+from django.urls import resolve, reverse
+
+
+class Enforce2FAMiddleware:
+    """Enforce 2FA middleware."""
+    def __init__(self, get_response):
+        """Initialize the middleware."""
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """Wrap around actual Admin calls."""
```

Comment:
This code won't limit itself to just the admin site as it makes no such check. Either code needs to do what the comment says or the comment needs changing.